vxquery-core/src/main/java/org/apache/vxquery/jsonparser/JSONParser.java

```diff
@@ -65,20 +66,43 @@ public JSONParser() throws JsonParseException {
         spStack = new ArrayList<UTF8StringPointable>();
         itemStack = new ArrayList<itemType>();
         svb = new StringValueBuilder();
+        sb = new SequenceBuilder();
         abvsStack.add(atomic);
         out = abvsStack.get(abvsStack.size() - 1).getDataOutput();
 
     }
 
-    public void parseDocument(File file, ArrayBackedValueStorage result) throws HyracksDataException {
+    public void atomicValues(int tag, JsonParser parser, DataOutput out, StringValueBuilder svb, int levelArray,
```

Comment:
Don't move this function. (Reduce the code changes.) My previous comment is nullified by the change to IParaser.